lib/product_list/product_list_page.dart

```diff
@@ -96,7 +96,10 @@ class _ProductListViewState extends State<ProductListView> {
                         : const ProductListItemShimmer();
                   }
 
-                  return ProductListItem(product: state.products[i]);
+                  return ProductListItem(
+                    key: Key(state.products[i].code),
+                    product: state.products[i],
+                  );
```

Comment:
Hmmm why that? 🤔

Reply:
When sorting the products, they are not rebuilt, and so the `existsInCart` stays true based on the index in the list.